lib/hacluster.pm

```diff
@@ -1137,4 +1138,38 @@ sub check_iscsi_failure {
     }
 }
 
+=head3 cluster_status_matches_regex
+
+Remotely check 'crm status' output against regular expression
+=over 1
+
+=item B<SHOW_CLUSTER_STATUS> - Output from 'crm status' command
+
+=back
+=cut
+
+sub cluster_status_matches_regex {
+    my ($show_cluster_status) = @_;
+    my @resource_list = ();
+    my $previous_line = '';
+    croak 'No crm status output' if (!$show_cluster_status);    
```

Comment:
you can use `unless` in place of `if (!`

Reply:
maybe the input validation can be moved even before the definition of `resource_list` and `previous_line`